### ISSUE TITLE
Single-session predictor core

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#141]
### Context

This slice replaces the dual-session data model with a single required `session` across the
predictor hierarchy in one atomic change, reaching GREEN-real for `predict` / `explain` /
`predict_with_uncertainty` through the public predictor APIs. It removes the unused deterministic
session that `MCDropoutPredictor` paid for, deletes the now-impossible "no mcd session"
`RuntimeError`, and drops the vestigial `ModelMetadata.mc_dropout` flag. Requires the skeleton
(Issue #1). Blocks the verb registry (Issue #3).

Key semantic point for MC-Dropout: today `MCDropoutPredictor.predict_with_uncertainty` reads
**only** `self._state.mcd_session` (the stochastic model) — the deterministic session in state is
never used by it. After this change, the single `session` for an `MCDropoutPredictor` **is** the
stochastic model, so every uncertainty call runs over a model whose logits vary per pass.

### User story

As an **inference service operator**, I want every predictor to load exactly one ONNX session so
that **MC-Dropout workers stop paying ResNet-50-sized load cost for a session they never use**.

### Acceptance criteria

- [ ] Given a valid deterministic ONNX model path, loading a `Classifier` or `Explainer` and calling its verb (`predict` / `explain`) returns the correct result type whose probability keys equal the model's classes.
- [ ] Given a valid stochastic (MC-Dropout) ONNX model path, loading an `MCDropoutPredictor` and calling `predict_with_uncertainty` returns an `UncertaintyResult` whose `mean_probabilities` sum to 1.0, whose keys equal the model's classes, whose `std_per_class` is non-zero across passes, and whose `n_passes` equals the requested count.
- [ ] Loading any predictor from a valid path constructs exactly one ONNX session, regardless of predictor type (observed by spying the `onnxruntime.InferenceSession` constructor).
- [ ] Loading a predictor from a path that does not exist raises `FileNotFoundError`.
- [ ] Calling `predict_with_uncertainty` on any successfully loaded `MCDropoutPredictor` never raises a "no MC-Dropout session" error — the load-time requirement of a second model is gone.
- [ ] A `Classifier` loaded via `from_registry` / `from_selector` (registry/wandb boundary mocked) yields the same class keys as the same model loaded via `from_path`.
- [ ] Constructing a `ModelMetadata` requires only `classes`, `input_shape`, `cam_target_layer`, `output_names`; there is no `mc_dropout` field to set, and constructing one with `mc_dropout=` raises `TypeError`.
- [ ] mypy clean; pytest green.

##### `src/radiologist/inference/models.py`

```python
@dataclass(frozen=True)
class ModelMetadata:
    classes: List[str]
    input_shape: List[int]
    cam_target_layer: str
    output_names: List[str]
    # contract: mc_dropout field removed
```

##### `src/radiologist/inference/base_predictor.py`

```python
@dataclass
class _PredictorState:
    session: ort.InferenceSession
    metadata: Dict[str, str]
    model_metadata: ModelMetadata
    # contract: det_session/mcd_session removed; single required session

class BasePredictor:
    @classmethod
    def from_path(cls, model_path: str) -> "BasePredictor":
        # contract: one ort.InferenceSession(model_path); FileNotFoundError when absent
    @classmethod
    def from_registry(cls, artifact_path, local_dir, registry=None) -> "BasePredictor":
        # contract: pull then from_path(model_path=<pulled>); signature unchanged
    @classmethod
    def from_selector(cls, selector, local_dir, registry=None) -> "BasePredictor":
        # contract: resolve+pull then from_path(model_path=<pulled>); signature unchanged

def _parse_model_metadata(metadata: Dict[str, str]) -> ModelMetadata:
    # contract: mc_dropout parameter removed
```

### Technical notes

Exact source edits (current line numbers):

- `models.py:65` — delete the `mc_dropout: bool` field from `ModelMetadata`.
- `base_predictor.py:53-58` — replace `_PredictorState` (`det_session`, `mcd_session: Optional[...] = field(default=None)`) with a single required `session: ort.InferenceSession` (keep `metadata` and `model_metadata`, keep the dataclass non-frozen).
- `base_predictor.py:66-100` — `from_path(cls, det_path, mcd_path=None)` → `from_path(cls, model_path: str)`. Body: `session = ort.InferenceSession(model_path)`; `metadata = _read_metadata(session)`; `model_metadata = _parse_model_metadata(metadata)`; construct `_PredictorState(session=session, metadata=metadata, model_metadata=model_metadata)`. Delete the `mcd_path`/`mcd_session` branch. Update the docstring (`det_path`/`mcd_path` → `model_path`).
- `base_predictor.py:128` and `:152` — `return cls.from_path(det_path=det_path)` → `return cls.from_path(model_path=<pulled>)` (rename local `det_path` → `model_path` for clarity; behaviour unchanged).
- `base_predictor.py:174-190` — `_parse_model_metadata(metadata, mc_dropout)` → `_parse_model_metadata(metadata)`; drop the `mc_dropout=mc_dropout` kwarg in the `ModelMetadata(...)` construction and the `mc_dropout` docstring arg.
- `classifier.py:68` — `session = self._state.det_session` → `session = self._state.session`.
- `explainer.py:66` — `session = self._state.det_session` → `session = self._state.session`.
- `mc_dropout.py:56-64` — replace `mcd_session = self._state.mcd_session` + the `if mcd_session is None: raise RuntimeError(...)` guard (lines 57-61) with `session = self._state.session`; the final call becomes `mc_dropout_predict(session, arr, n_passes=n_passes)`. Update the docstring to drop the `Raises: RuntimeError` note.

Test updates (owned by this slice):

- `tests/conftest.py:49-63` — collapse `predictor_with_mcd` and `predictor_without_mcd` into a single `predictor` fixture: `MCDropoutPredictor.from_path(model_path=build_mcd_onnx(tmp_path))`. Use the **stochastic** `build_mcd_onnx` model so `std_per_class` is non-zero (a deterministic `build_det_onnx` model would make the std-nonzero assertion flaky/false).
- `tests/test_mc_dropout.py` — update the `TestPredictWithUncertainty` fixtures from `predictor_with_mcd` to `predictor`; delete `test_raises_runtime_error_when_no_mcd_model` (lines 84-89) — that error path no longer exists. `TestMcDropoutPredict` (the stateless helper tests, lines 97-131) is unchanged.
- `tests/test_classifier.py` and `tests/test_explainer.py` — rename every `from_path(det_path=…)` keyword call to `from_path(model_path=…)`. Leave the internal `_FakeRegistry`/`_FakeSelectorRegistry` constructor params (their own `det_path` attributes) alone — they are unrelated helper internals, not the `from_path` API.
- `tests/test_app.py:45-61` — rename `Classifier.from_path(det_path=…)` / `Explainer.from_path(det_path=…)` to `model_path=`; change the `mcd_predictor` fixture (lines 57-61) to load a single stochastic model: `MCDropoutPredictor.from_path(model_path=build_mcd_onnx(tmp_path, filename="mcd.onnx"))` (drop the det+mcd two-arg call).
- `tests/test_public_api.py:114-121` — drop `mc_dropout=False` from the `ModelMetadata(...)` construction in `test_result_dataclasses_are_importable`.

README update (owned by this slice):

- `radiologist-inference/README.md` — update the API examples and table: `from_path(det_path="model.onnx")` → `from_path(model_path="model.onnx")`; remove the `mcd_path` example (lines ~51-53) and the `# MC-Dropout uncertainty (requires mcd_path)` comment; update the `from_path` signature row to `(model_path: str) -> BasePredictor`; drop "requires `mcd_path` at load time" from the `predict_with_uncertainty` row; drop `mc_dropout: bool` from the `ModelMetadata` field row.

### Design notes

The single field is named `session` (not `det_session` or `inference_session`) because after this
change there is exactly one session per predictor and its "deterministic vs stochastic" character
is a property of the loaded artifact, not of the state slot — the state is verb-agnostic. Keeping
`_PredictorState` a plain (non-frozen) dataclass preserves the existing `cls.__new__()` +
attribute-assignment construction pattern in `from_path`; introducing a frozen state or a builder
here would be churn beyond the epic's purpose. Removing `ModelMetadata.mc_dropout` rather than
retaining it as always-`False` avoids leaving a dead field that future readers would mistake for a
live capability flag.
